### PR TITLE
ci: automate beta release workflow

### PR DESCRIPTION
**Pull request by ckrook**
## **Describe pull-request**  
This pull request introduces a new GitHub Actions workflow named "Beta Release". The workflow automates the process of publishing beta versions of our packages to NPM, specifically for the Core, Angular, Angular-17, and React packages. It allows for manual triggering with customizable inputs.

## **Issue Linking:** 
_Choose one of the following options_
- **Jira:**  `ci(beta-release)`: [CDEP-3561](https://tegel.atlassian.net/browse/CDEP-3561)    

## **How to test**  
To test this you will need to run it with Act since its not merged in develop yet the workflow does not exist within the Github UI.

Run this command with Act:
`act -W .github/workflows/beta-release.yml -e event.json -P ubuntu-latest=catthehacker/ubuntu:act-latest -s NPM_TOKEN=XXXXXXXXXXXXXX -s GITHUB_TOKEN=XXXXXXXXXXXXXX`

Also add an event.json file in the root of the project wit the following code:
```
{
  "ref": "refs/heads/ci/beta-release-workflow",
  "inputs": {
    "branch": "ci/beta-release-workflow",
    "nodeVersion": "20.9.0",
    "betaVersionSuffix": "betaworkflow.7",
    "includeAngular17": "false",
    "includeAngular": "true",
    "includeReact": "true"
  }
}
```

## **Checklist before submission**
Note: Since this PR pertains to the CI/CD pipeline and does not affect the UI or user-interactive components, UI testing steps are not applicable.
- [ ] I have added unit tests for my changes (if applicable)
- [ ] All existing tests pass
- [ ] I have updated the documentation (if applicable)
- [ ] Not breaking production behavior
- [ ] Behavior available in storybook with documented descriptions (if applicable)
- [ ] `npm run build-all` without errors

## **Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

## **Screenshots**  
_Include before/after screenshots for UI changes._

## **Additional context**  
_Add any other context or feedback requests about the pull-request here._

[CDEP-3561]: https://tegel.atlassian.net/browse/CDEP-3561?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/beta-release.yml [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+jobs:
+  beta-release-core:
+    runs-on: ubuntu-latest
+    env:
+      WORKING_DIRECTORY: packages/core
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          ref: develop
```
[the remark this answers is not in the document]
Comment (theJohnnyMe) [reply .github/workflows/beta-release.yml:54]: I agree with @mistermalm.

[the remark this answers is not in the document]
Comment (ckrook) [reply .github/workflows/beta-release.yml:54]: Good points! 🔥

**Comment by timrombergjakobsson** [comment]
@ckrook don't forget the conventional commit formatting of the title!

**Review by theJohnnyMe — APPROVED** [review]
LGTM, lets give it a shot ;)
Nice work @ckrook

File: .github/workflows/beta-release.yml [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+      betaVersionSuffix:
+        description: 'Beta version suffix (e.g., beta.0, beta.1)'
+        required: true
+        default: 'beta.0'
+        type: string
+
+      includeAngular17:
+        description: 'Include Angular 17 package?'
+        required: false
+        default: 'true'
```
Comment (timrombergjakobsson) [inline .github/workflows/beta-release.yml:26]:
```suggestion
        default: 'false'
```
? Feels like it should be false as default, best practice?

File: .github/workflows/beta-release.yml [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+        type: choice
+        options:
+          - 'true'
+          - 'false'
+
+      includeAngular:
+        description: 'Include Angular package?'
+        required: false
+        default: 'true'
```
Comment (timrombergjakobsson) [inline .github/workflows/beta-release.yml:35]:
```suggestion
        default: 'false'
```

File: .github/workflows/beta-release.yml [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+        type: choice
+        options:
+          - 'true'
+          - 'false'
+
+      includeReact:
+        description: 'Include React package?'
+        required: false
+        default: 'true'
```
Comment (timrombergjakobsson) [inline .github/workflows/beta-release.yml:44]:
```suggestion
        default: 'false'
```

**Review by mistermalm — APPROVED** [review]

**Pull request merged**